src/persist/src/indexed/mod.rs

```diff
@@ -319,19 +386,32 @@ impl<U: Buffer, L: Blob> Indexed<U, L> {
             // future's seqno_upper and that there is nothing for that future in
             // [future.seqno_upper, self.futures_seqno_upper). Use this to make the
             // seqnos of all the future batches line up.
-            debug_assert_eq!(desc.start, self.futures_seqno_upper);
+
+            // We cannot assert strict equality here because a buffer may have
+            // previously failed to truncate.
+            debug_assert!(desc.start <= self.futures_seqno_upper);
             let new_start = future.seqno_upper()[0];
-            debug_assert!(new_start <= desc.start);
+            debug_assert!(new_start <= self.futures_seqno_upper);
```

Comment:
i'm probably just being dense, why this change?

Reply:
I think there's still something a bit funky here, but most of this gets deleted in the next PR so, let's just address what's left there